components/web/Download/component-sdk-v2

```diff
@@ -0,0 +1,25 @@
+name: Download data (KFP SDK v2)
+description: Downloads data from the specified URL. (Updated for KFP SDK v2.)
+inputs:
+- {name: Url, type: String}
+- {name: curl options, type: String, default: '--location', description: 'Additional options given to the curl bprogram. See https://curl.haxx.se/docs/manpage.html'}
+outputs:
+- {name: Data, type: Artifact}
```

Comment:
Are you sure this will be compatible with various downstream component inputs with types like Dataset or Model?

Reply:
From the testing I have done, it doesn't look like you can currently go from an Artifact to a Dataset without another component to translate. Since we don't know what is getting downloaded, I suggested Artifact in this case.

Reply:
What if we do not specify any type? Untyped output is considered compatible with any inputs. This looks like a valid case for the downloader components.

Reply:
SGTM, I've updated the component and tested it with the docs sample.

Reply:
@chensun can we change type checking so `Artifact` behaves like `Any` type?

Reply:
> can we change type checking so Artifact behaves like Any type?
Yes, we can make `Artifact` be compatible with any artifact type. I'll send a PR shortly.